tests/epyccel/test_epyccel_variable_annotations.py

```diff
@@ -229,13 +229,15 @@ def test_homogeneous_list_annotation_int(stc_language):
     def homogeneous_list_annotation():
         a: 'list[int]'  # pylint: disable=unused-variable
         a = [1, 2, 3, 4]
+        return a[0], a[1], a[2], a[3]
     epyc_homogeneous_list_annotation = epyccel(homogeneous_list_annotation, language=stc_language)
     assert epyc_homogeneous_list_annotation() == homogeneous_list_annotation()
     assert isinstance(epyc_homogeneous_list_annotation(), type(homogeneous_list_annotation()))
 
 def test_homogeneous_list_without_annotation(stc_language):
     def homogeneous_list():
         a = [1, 2, 3, 4] # pylint: disable=unused-variable
+        return a[0], a[1], a[2], a[3]
```

Comment:
I think these are the only tests for list initialization, if it is the case, don't you think that having a test for both initialization and index retrieval goes against the separationf of concerns principle?

Reply:
No. I don't. In theory I agree that it is good to separate concerns. However until it is possible to return a list (which will come quite late in development) the only way to validate the values saved in the list is to retrieve them via indices. If the following code compiles:
```python
    def homogeneous_list():
        a = [1, 2, 3, 4] # pylint: disable=unused-variable
```
there is no way to tell if it is translated to:
```c
void homogeneous_list() {
}
```